x-pack/plugins/security_solution/public/management/pages/event_filters/view/event_filters_list_page.test.tsx

```diff
@@ -0,0 +1,134 @@
+
+  const dataReceived = () =>
+    act(async () => {
+      await waitForAction('eventFiltersListPageDataChanged', {
+        validate(action) {
+          return isLoadedResourceState(action.payload);
+        },
+      });
+    });
+
+  beforeEach(() => {
+    const mockedContext = createAppRootMockRenderer();
+
+    ({ history, coreStart } = mockedContext);
+    render = () => (renderResult = mockedContext.render(<EventFiltersListPage />));
+    mockedApi = eventFiltersListQueryHttpMock(coreStart.http);
+    waitForAction = mockedContext.middlewareSpy.waitForAction;
+
+    act(() => {
+      mockedContext.setExperimentalFlag({ eventFilteringEnabled: true });
+      history.push('/event_filters');
+    });
+  });
+
+  describe('And no data exists', () => {
+    beforeEach(async () => {
+      mockedApi.responseProvider.eventFiltersList.mockReturnValue({
+        data: [],
+        page: 1,
+        per_page: 10,
+        total: 0,
+      });
+
+      render();
+
+      await act(async () => {
+        await waitForAction('eventFiltersListPageDataExistsChanged', {
+          validate(action) {
+            return isLoadedResourceState(action.payload);
+          },
+        });
+      });
+    });
+
+    it('should show the Empty message', () => {
+      expect(renderResult.getByTestId('eventFiltersEmpty')).toBeTruthy();
+      expect(renderResult.getByTestId('eventFiltersListEmptyStateAddButton')).toBeTruthy();
+    });
+
+    it('should open create flyout when add button in empty state is clicked', async () => {
+      act(() => {
+        fireEvent.click(renderResult.getByTestId('eventFiltersListEmptyStateAddButton'));
+      });
+
+      expect(renderResult.getByTestId('eventFiltersCreateEditFlyout')).toBeTruthy();
+      expect(history.location.search).toEqual('?show=create');
+    });
+  });
+
+  describe('And data exists', () => {
+    it('should show loading indicator while retrieving data', async () => {
+      let releaseApiResponse: () => void;
+
+      mockedApi.responseProvider.eventFiltersList.mockDelay.mockReturnValue(
+        new Promise((r) => (releaseApiResponse = r))
+      );
+      render();
+
+      expect(renderResult.getByTestId('eventFiltersContent-loader')).toBeTruthy();
+
+      const wasReceived = dataReceived();
+      releaseApiResponse!();
+      await wasReceived;
+
+      expect(renderResult.container.querySelector('.euiProgress')).toBeNull();
+    });
+
+    it('should show items on the list', async () => {
+      render();
+      await dataReceived();
+
+      expect(renderResult.getByTestId('eventFilterCard')).toBeTruthy();
+    });
+
+    it('should render expected fields on card', async () => {
+      render();
+      await dataReceived();
+
+      // Take a snapshot of the card to ensure that all of the optional fields we need
+      // for Event filters use case (modified info ++ name) are displayed as we expect it.
+      expect(renderResult.getByTestId('eventFilterCard')).toMatchSnapshot();
```

Comment:
As the eventFilterCard is a component from other team, can we just check that the text we want is displayed? I'm wondering if changes on the eventFilterCard (classNames, ids, etc) will break this test

Reply:
yeah, you right. I refactored it to instead grab the values and ensure they are displayed.